### PR TITLE
chore(ci): add full release workflow generation [skip ci]

### PR DESCRIPTION
**Pull request by Okhelifi**
## Issue

https://gravitee.atlassian.net/browse/APIM-2635

## Description

Add full release workflow generation

### CODE REVIEW

File: .circleci/ci/src/pipelines/tests/pipeline-full-release.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+import * as fs from 'fs';
+import { generateFullReleaseConfig } from '../pipeline-full-release';
+
+describe('Full release tests', () => {
+  it.each`
+    branch     | isDryRun | expectedResult
+    ${'4.2.x'} | ${true}  | ${'release-4-2-0-dry-run.yml'}
+    ${'4.2.x'} | ${false} | ${'release-4-2-0-no-dry-run.yml'}
+  `('should build full release config on $branch with dry run $isDryRun', ({ branch, isDryRun, expectedResult }) => {
```
Comment (jourdiw) [inline .circleci/ci/src/pipelines/tests/pipeline-full-release.spec.ts]: Is it possible that a release could be an alpha or that we may or may specify dockerTagAsLatest?

Comment (Okhelifi) [reply .circleci/ci/src/pipelines/tests/pipeline-full-release.spec.ts]: As it has been tested in other tests files I didn't add it but I can add it here too yes 👍🏼

Comment (Okhelifi) [reply .circleci/ci/src/pipelines/tests/pipeline-full-release.spec.ts]: Done

File: .circleci/ci/src/workflows/workflow-full-release.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        name: 'Announce release is starting',
+        message: `🚀 Starting APIM ${environment.graviteeioVersion} release!`,
+      }),
+
+      // APIM Portal
+      new workflow.WorkflowJob(webuiBuildJob, {
+        context: config.jobContext,
+        name: 'Build APIM Portal and publish image',
+        'apim-ui-project': 'gravitee-apim-portal-webui',
+        'docker-image-name': 'apim-portal-ui',
```
Comment (phiz71) [inline .circleci/ci/src/workflows/workflow-full-release.ts]:
```suggestion
        'docker-image-name': `${config.dockerImages.portal}`,
```

File: .circleci/ci/src/workflows/workflow-full-release.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        'apim-ui-project': 'gravitee-apim-portal-webui',
+        requires: ['Build APIM Portal and publish image'],
+      }),
+
+      // APIM Console
+      new workflow.WorkflowJob(webuiBuildJob, {
+        context: config.jobContext,
+        name: 'Build APIM Console and publish image',
+        'apim-ui-project': 'gravitee-apim-console-webui',
+        'docker-image-name': 'apim-management-ui',
```
Comment (phiz71) [inline .circleci/ci/src/workflows/workflow-full-release.ts]:
```suggestion
        'docker-image-name': `${config.dockerImages.console}`,
```

File: .circleci/ci/src/workflows/workflow-full-release.ts (hunk 2 above)
Comment (Okhelifi) [reply .circleci/ci/src/workflows/workflow-full-release.ts]: Done

File: .circleci/ci/src/workflows/workflow-full-release.ts (hunk 3 above)
Comment (Okhelifi) [reply .circleci/ci/src/workflows/workflow-full-release.ts]: Done

**Review by phiz71 — APPROVED** [review]

**Pull request merged**